features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/stored-procedures/oracle/sessiondata-cleanup/oracle-sessiondata-cleanup.sql

```diff
@@ -200,13 +159,155 @@ BEGIN
+                    COMMIT;
+                END IF;
+                -- End of deleting user-session mappings from 'IDN_AUTH_USER_SESSION_MAPPING' table
+
+                -- Deleting session app info from 'IDN_AUTH_SESSION_APP_INFO' table
+                SELECT COUNT(*) INTO ROWCOUNT from ALL_TABLES where OWNER = CURRENT_SCHEMA AND table_name = upper('IDN_AUTH_SESSION_APP_INFO');
+                IF (ROWCOUNT = 1) then
+                    EXECUTE IMMEDIATE 'DELETE FROM IDN_AUTH_SESSION_APP_INFO WHERE SESSION_ID IN (SELECT SESSION_ID FROM TEMP_USER_SESSION_BATCH)';
+                    sessionappinfocleanupcount := SQL%rowcount;
+                    COMMIT;
+                    IF ( tracingenabled ) THEN
+                        deletedsessionapppinfo := deletedsessionapppinfo + sessionappinfocleanupcount;
+                        EXECUTE IMMEDIATE 'INSERT INTO LOG_WSO2_SESSION_CLEANUP_SP (TIMESTAMP,LOG) VALUES (TO_CHAR( SYSTIMESTAMP, ''DD.MM.YYYY HH24:MI:SS:FF4''),''DELETED SESSION APP INFO COMPLETED WITH '|| sessionappinfocleanupcount||''')';
+                        COMMIT;
+                    END IF;
+                END if;
+                -- End of deleting session app info from 'IDN_AUTH_SESSION_APP_INFO' table
+
+                -- Deleting session metadata from 'IDN_AUTH_SESSION_META_DATA' table
+                SELECT COUNT(*) INTO ROWCOUNT from ALL_TABLES where OWNER = CURRENT_SCHEMA AND table_name = upper('IDN_AUTH_SESSION_META_DATA');
+                IF (ROWCOUNT = 1) then
+                    EXECUTE IMMEDIATE 'DELETE FROM IDN_AUTH_SESSION_META_DATA WHERE SESSION_ID IN (SELECT SESSION_ID FROM TEMP_USER_SESSION_BATCH)';
+                    sessionmetadatacleanupcount := SQL%rowcount;
+                    COMMIT;
+                    IF ( tracingenabled ) THEN
+                        deletedsessionmetadata := deletedsessionmetadata + sessionmetadatacleanupcount;
+                        EXECUTE IMMEDIATE 'INSERT INTO LOG_WSO2_SESSION_CLEANUP_SP (TIMESTAMP,LOG) VALUES (TO_CHAR( SYSTIMESTAMP, ''DD.MM.YYYY HH24:MI:SS:FF4''),''DELETED SESSION METADATA COMPLETED WITH '|| sessionmetadatacleanupcount||''')';
+                        COMMIT;
+                    END IF;
+                END if;
+                -- End of deleting session metadata from 'IDN_AUTH_SESSION_META_DATA' table
+
+                EXECUTE IMMEDIATE 'DELETE FROM IDN_AUTH_USER_SESSION_TMP WHERE SESSION_ID IN (SELECT SESSION_ID FROM TEMP_USER_SESSION_BATCH)';
+                COMMIT;
+
+                IF ( tracingenabled ) THEN
+                EXECUTE IMMEDIATE 'INSERT INTO LOG_WSO2_SESSION_CLEANUP_SP (TIMESTAMP,LOG) VALUES (TO_CHAR( SYSTIMESTAMP, ''DD.MM.YYYY HH24:MI:SS:FF4''),''REMOVED THE BATCH FROM TABLE IDN_AUTH_USER_SESSION_TMP'')';
+                COMMIT;
+                END IF;
+
+                EXECUTE IMMEDIATE 'DELETE FROM IDN_AUTH_EXPIRED_SESSION_TMP WHERE SESSION_ID IN (SELECT SESSION_ID FROM TEMP_USER_SESSION_BATCH)';
```

Comment:
use inner join

Reply:
Resolving due to the same reason as #3313 (comment)